dips/dip-158.md

```diff
@@ -0,0 +1,794 @@
+
+---
+# Abstract / Motivation
+
+
+This DIP, as described here, is an extension of the Off-Chain Protocol to support more advanced payment functionality - particularly eCommerce P2M payments. 
+
+## How P2M Payments Different than P2P Transfers 
+There are some key differences in the process of paying a merchant (P2M) that differ from simple P2P transfer:
+
+|      |P2M|P2P|
+|------|---|---|
+|**Payer Data**|Buyer details are required for **every** payment / purchase in order to perform risk checks and apply AML policies|Required only for transfers for which Travel Rule applies. i.e., over 1000 USD|
+|**Payment Owner**|The merchant, i.e. the receiver of funds, "drives" the process and controls the type of payment and its timing. For example, the merchant will decide when to capture the funds and whether to capture the entire amount or only partial amount |The payer, i.e. the sender of funds, determines the nature of the transfer|
+|**Reconciliation**|The merchant requires a way to correlate the transfer of funds to a payment or payment request for reconciliation purposes|No reconciliation|
+|
+
+## P2M Payment Types
+There are two types of P2M payments described in this document: 
+
+1. **Charge**. Charge payment stipulates an immediate capture of funds. This means that a successful off-chain negotiation for such payment will result in funds being captured 
+2. **Auth / Capture**. Auth / Capture payment, as the name suggests, means that the payment is divided to two steps: Authorization and Capture. Successful Authorization implies that the funds are locked so they can later be captured
+
+This DIP defines the type of eCommerce P2M payments, corresponding requests and the payment structures. 
+
+
+---
+# Synonyms
+
+For the sake of simplicity, we use the following terms:
+* "Sender", "Payer", "Wallet", "Buyer VASP" or "User VASP" - The user's wallet. It is the VASP representing the customer/buyer. This is the VASP that will send the funds and eventually will submit the on-chain transaction to the blockchain
+* "Receiver", "Receiving VASP" or "Merchant VASP" - The VASP representing the merchant which receives the funds (transaction)
+* "Buyer", "Customer" or "User" - The person who wishes to pay the merchant for goods or services
+
+
+
+---
+# Pre-requisite: Sharing Payment Identifier and Receiver Address
+
+A pre-requisite for the off-chain negotiations is that both parties (sending and receiving VASPs) have managed to share a common payment identifier and the receiver address is known. This DIP specifies a couple of examples for exchanging identifiers between VASPs, however,  there could be others.
+
+See appendix A
+
+
+---
+# P2M Off-Chain Protocol Specification
+This section, including all its sub-sections, specifies the P2M payment types within the scope of this document, the required structures, payloads and the expected sequence and of calls based on this protocol.
+
+Content:
```

Comment:
1. These should be hyperlinks into the corresponding parts.
2. I wouldn't use italics font here, it would typically indicate a new definition or emphasis

Reply:
1. Correct. I will hyperlinks where required
2. Will remove the Italics